independent-projects/qute/core/src/main/java/io/quarkus/qute/HtmlEscaper.java

```diff
@@ -8,6 +8,51 @@
 public class HtmlEscaper extends CharReplacementResultMapper {
 
     private final List<String> escapedContentTypes;
+    // NOTE: this holding 8 values to allow the JIT to remove the bound checks since
+    // the replacement id is always in range [0, 7] due to the REPLACEMENT_ID_MASK value
+    private static final String[] REPLACEMENTS = { null, "&quot;", "&#39;", "&amp;", "&lt;", "&gt;", null, null };
+    // We use 4 bits to pack the replacement id for each Latin character in the [0, 255] range.
```

Comment:
The doc here is not up to date with the latest changes..I ended up accepting having a slightly bigger footprint, but better latency